to-square-or-not-no-square.rb

```diff
@@ -0,0 +1,13 @@
+#The function takes square root if the number has an integer square root, otherwise square the number.
+
+#square_or_square_root([4,3,9,7,2,1]) 
+
+#-> [2,9,3,49,4,1]
+
+def square_or_square_root(arr)
+  arr.map do |num|
+    sqrt = num**0.5
+    sqrt % 1 == 0 ? num**0.5 : num**2
```

Comment:
as an alternative you can use more OOP approach asking object whether it is Integer or not.
```
irb(main):005:0> 2.2.is_a? Integer
=> false
irb(main):006:0> 2.is_a? Integer
=> true
```